core/modules/comment/comment.module

```diff
@@ -2311,38 +2309,74 @@ function comment_action_info() {
       'type' => 'comment',
       'callback' => 'comment_unpublish_action',
     ),
+    'comment_delete_action' => array(
+      'label' => t('Delete comment'),
+      'type' => 'comment',
+      'callback' => 'comment_delete_action',
+    ),
   );
 }
 
 /**
  * Publishes a comment.
  *
- * @param $comment
- *   The comment object.
+ * @param Comment $comment
+ *   A comment entity.
+ * @param $context
+ *   Contextual information about the triggered action.
  *
  * @ingroup actions
  */
-function comment_publish_action($comment, $context = array()) {
+function comment_publish_action(Comment $comment, $context = array()) {
   $comment->status = COMMENT_PUBLISHED;
   $comment->save();
 }
 
 /**
  * Unpublishes a comment.
  *
- * @param $comment
- *   An optional comment object.
- * @param array $context
- *   Array with components:
+ * @param Comment $comment
+ *   A comment entity.
+ * @param $context
+ *   Contextual information about the triggered action.
  *   - 'cid': Comment ID. Required if $comment is not given.
  *
  * @ingroup actions
  */
-function comment_unpublish_action($comment, $context = array()) {
+function comment_unpublish_action(Comment $comment, $context = array()) {
```

Comment:
Are these changes needed?

Reply:
Yes, we've been casting entities as their classes in core, such as `Node $node`.

Reply:
In old code as well as new? I'm just asking because this will make D7 patches less likely to apply and I haven't seen it being done anywhere else.

Reply:
Yes. E.g.:

```
function comment_save(Comment $comment) {
  return $comment->save();
}
```